gtsfm/loader/loader_base.py

```diff
@@ -183,6 +183,13 @@ def get_camera_intrinsics(self, index: int) -> Cal3Bundler:
             u0=intrinsics_full_res.px() * scale_u,
             v0=intrinsics_full_res.py() * scale_v,
         )
+
+        if rescaled_intrinsics.fx() <= 0:
+            raise RuntimeError("Focal length must be positive.")
+
+        if rescaled_intrinsics.px() <= 0 or rescaled_intrinsics.py() <= 0:
+            raise RuntimeError("Principal point must have positive coordinates.")
```

Comment:
I think these checks should happen above, for intrinsics_full_res - and `get_camera_intrinsics_full_res` would be the culprit.

Reply:
good catch, will update

Reply:
Updated now